app/src/main/java/org/android/go/sopt/ui/main/MainActivity.kt

```diff
@@ -1,23 +1,73 @@
 package org.android.go.sopt.ui.main
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
-import org.android.go.sopt.Week1Application
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.RecyclerView
+import org.android.go.sopt.R
+import org.android.go.sopt.binding.BindingActivity
 import org.android.go.sopt.databinding.ActivityMainBinding
-import org.android.go.sopt.util.HOBBY_KEY
-import org.android.go.sopt.util.NAME_KEY
+import org.android.go.sopt.ui.main.gallery.GalleryFragment
+import org.android.go.sopt.ui.main.home.HomeFragment
+import org.android.go.sopt.ui.main.search.SearchFragment
 
-class MainActivity : AppCompatActivity() {
-    private lateinit var binding: ActivityMainBinding
+class MainActivity : BindingActivity<ActivityMainBinding>(R.layout.activity_main) {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        binding = ActivityMainBinding.inflate(layoutInflater)
-        setContentView(binding.root)
 
-        val name = Week1Application.prefs.getString(NAME_KEY, "nothing")
-        val hobby = Week1Application.prefs.getString(HOBBY_KEY, "nothing")
-        binding.tvName.append(name)
-        binding.tvHobby.append(hobby)
+        loadHomeFragment()
+        initBnvItemSelectedListener()
+        initBnvItemReselectedListener()
+    }
+
+    private fun initBnvItemReselectedListener() {
+        binding.bnvMain.setOnItemReselectedListener {
+            when(supportFragmentManager.findFragmentById(R.id.fcv_main)){
+                is HomeFragment -> {
+                    val recyclerView = findViewById<RecyclerView>(R.id.rv_home)
+                    recyclerView.scrollToPosition(0)
+                }
+                is GalleryFragment -> {
+                    val recyclerView = findViewById<RecyclerView>(R.id.rv_gallery)
+                    recyclerView.scrollToPosition(0)
+                }
+            }
+        }
+    }
+
+    private fun initBnvItemSelectedListener() {
+        binding.bnvMain.setOnItemSelectedListener { item ->
+            when (item.itemId) {
+                R.id.home_menu -> {
+                    changeFragment(HomeFragment())
+                    true
+                }
+                R.id.gallery_menu -> {
+                    changeFragment(GalleryFragment())
+                    true
+                }
+                R.id.search_menu -> {
+                    changeFragment(SearchFragment())
+                    true
+                }
+                else -> false
+            }
+        }
+    }
+
+    private fun loadHomeFragment() {
+        val currentFragment = supportFragmentManager.findFragmentById(R.id.fcv_main)
+        if (currentFragment == null) {
+            supportFragmentManager.beginTransaction()
+                .add(R.id.fcv_main, HomeFragment())
+                .commit()
+        }
```

Comment:
```suggestion
        if (currentFragment == null) changeFragment(HomeFragment())
```

여기서도 `changeFragment` 함수를 활용할 수 있을 것 같습니다!